src/platform/silabs/wifi/SiWx/WifiInterfaceImpl.cpp

```diff
@@ -837,6 +837,8 @@ sl_status_t WifiInterfaceImpl::TriggerPlatformWifiDisconnection()
 #if CHIP_CONFIG_ENABLE_ICD_SERVER
 CHIP_ERROR WifiInterfaceImpl::ConfigurePowerSave(PowerSaveInterface::PowerSaveConfiguration configuration, uint32_t listenInterval)
 {
+    VerifyOrReturnError(currentPowerSaveConfiguration != configuration, CHIP_NO_ERROR);
```

Comment:
- I would not recommend using this here as it is to return errors. Lets use a simple if may be.
- I remember you mentioning about the counter which can be used? Just trying to understand if this new variable is required or we can use some existing variables to achieve the same.